filters.ml

```diff
@@ -1066,6 +1088,7 @@ let run com tctx main =
 			captured_vars com;
 			promote_complex_rhs com;
 			if com.config.pf_add_final_return then add_final_return else (fun e -> e);
+			if com.platform = Js then wrap_js_exceptions com else (fun e -> e);
```

Comment:
I wanted to run that before the DCE, so HaxeError isn't generated if we don't throw non-`Error` exceptions.

Actually, I think we should have more platform-specific filters that run before the dce, so we can remove unnecesarry `@:keep`s and make generated code much cleaner. 
What I'm not sure about is how to organize that without making a second gencommon.